src/app/_fetchers/AssigneeFetcher.tsx

```diff
@@ -29,8 +35,7 @@ const fetchAssignee = async (token: string, userType?: UserType): Promise<IAssig
   return (await res.json()).users as IAssignee
 }
 
-export const AssigneeFetcher = async ({ token, userType }: Props) => {
+export const AssigneeFetcher = async ({ token, userType, viewSettings }: Props) => {
   const assignableUsersWithType = addTypeToAssignee(await fetchAssignee(token, userType))
-
-  return <ClientSideStateUpdate assignee={assignableUsersWithType} />
+  return <ClientSideStateUpdate assignee={assignableUsersWithType} viewSettings={viewSettings} />
```

Comment:
I don't understand this. viewSettings is being fetched in page itself, isn't it? And so it should have been already set there.

Reply:
The problem here is, viewsettings is updated in page itself which has dependencies in the filter by assignee button list too. For example, if client tasks tab is saved in viewsetting for a user, the filter by assignee button should show only client and company as options. Viewsettings being updated way before assignee list gets updated causes all the assigne to show in the filter by assignee options regardless the tab saved in viewsettings. For this purpose, viewSettings needs to be updated once agains when assignee list gets updated.